source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -902,7 +903,80 @@ namespace AdaptiveCards { namespace Rendering { namespace Uwp
 
                 HString title;
                 THROW_IF_FAILED(action->get_Title(title.GetAddressOf()));
-                XamlHelpers::SetContent(button.Get(), title.Get());
+                ComPtr<IUriRuntimeClass> iconUrl;
+                THROW_IF_FAILED(action->get_IconUrl(iconUrl.GetAddressOf()));
+                // Check if the button has an iconUrl
+                if (iconUrl.Get()) 
+                {
```

Comment:
>{ [](start = 16, length = 1)

This function is already way too big. Can you pull the stuff under this if into a helper function?